src/cobertura.d

```diff
@@ -0,0 +1,199 @@
+module cobertura;
+
+import std.typecons : Nullable, nullable;
+import std.traits : isSomeString, isCallable;
+import std.range : isInputRange, ElementType;
+
+struct LineCoverage {
+	size_t index;
+	size_t count;
+}
+
+struct FileCoverage {
+	string path;
+	LineCoverage[] lines;
+	size_t linesCovered;
+}
+
+FileCoverage parseLstFile(string path) {
+	import std.stdio : File;
+	return parseLstFile(File(path).byLine, path);
+}
+
+template tryTo(T) {
+	Nullable!T tryTo(R)(R r) {
+		import std.conv : to;
+		try {
+			return nullable(r.to!T);
+		} catch (Exception e) {
+			return Nullable!T.init;
+		}
+	}
+}
+
+enum isNullable(T) = is(T : Nullable!P, P);
+
+auto filterNullable(Range)(Range r) if (isNullable!(ElementType!Range)) {
```

Comment:
the name is misleading. I would say it is getIfNotNull?

But as it is called once inline filter!(i => !i.isNull()).map!(i => i.get()) should have worked as well

Reply:
You are right. It filters *non-Nullable*, not Nullable. And then it unpacks them. Haskell probably has a name for it.

Reply:
especially, as you could run the old thing by typing dmd -run source/app.d

Reply:
Haskell calls it [`catMaybes`](https://hackage.haskell.org/package/base-4.7.0.2/docs/Data-Maybe.html#v:catMaybes)

Scala calls it `flatten`.

I'll go with `flatten`

Reply:
Sure.